### ISSUE TITLE
Repo: Rename primary branch

### ISSUE THREAD

**Issue by kraftbj** [Automattic/jetpack#18946]
Supports #16099

Now that GitHub [has native tooling for renaming the primary branch](https://github.com/github/renaming#renaming-existing-branches), it's time for us to move from `master` to `trunk` (WP uses `trunk` for SVN and what WP and a8c have been moving toward).

Using this issue to discuss what all we need to be sure to ensure is covered.

In terms of timing, execute the change immediately following a release branch cut?

- [ ] Fusion?
- [ ] Actions
- [ ] Beta builder
- [ ] Mirror repos -- do we change all of these at the same time?
- [ ] What else? (feel free to edit this to add more items)

What y'all think @jeherve, @anomiex?

**Comment by anomiex** [comment]
> Mirror repos -- do we change all of these at the same time?

Yes. It'd be easier to do that than to add code to swap branch names in the middle of mirroring.

It's not clear to me from GH's docs whether a post-rename push to "master" will fail or will be forwarded with a warning, but either way that seems like the best we can do for handling the in-between state.

I'd probably update the mirroring actions to add "trunk" to the list of branches in all the actions where it's checked for without removing "master", then rename the mirrors, then the monorepo, then remove "master" from the lists of branches in actions. The double patches reduce the risk of missing workflow runs breaking something in the in-between state. But if we don't want to be quite so cautious, we could get away with merging a single patch.

**Comment by kraftbj** [comment]
I'm happy for a double-patch to be safer.

**Comment by coder-karen** [comment]
Closing this as the primary branch has now been renamed from `master` to `trunk` (including changes in the Field Guide, MC, etc).

p9dueE-59T-p2
